consensus/istanbul/core/handler.go

```diff
@@ -121,11 +121,17 @@ func (c *core) handleEvents() {
 					//c.backend.Gossip(c.valSet, p)
 				}
 			}
-		case _, ok := <-c.timeoutSub.Chan():
-			if !ok {
+		case ev, ok := <-c.timeoutSub.Chan():
+			if !ok || ev.Data == nil {
+				logger.Warn("Invalid message type from timeout channel")
+				return
+			}
+			data, ok := ev.Data.(timeoutEvent)
+			if !ok || data.view == nil {
+				logger.Warn("Invalid message from timeout channel", "msg", ev.Data)
```

Comment:
Maybe the logs are written backward?
```suggestion
				logger.Warn("Invalid message from timeout channel")
				return
			}
			data, ok := ev.Data.(timeoutEvent)
			if !ok || data.view == nil {
				logger.Warn("Invalid message type from timeout channel", "msg", ev.Data)
```